tm2/pkg/bft/consensus/replay.go

```diff
@@ -306,6 +307,7 @@ func (h *Handshaker) ReplayBlocks(
 		if err != nil {
 			return nil, err
 		}
+		saveGenesisRes(h.stateDB, &res)
```

Comment:
> not sure if this is possible, but worth checking out
block 0 is treated specially in complex procedures, I tried a bit this approach then fell back on the one in this PR, I'll try again though

Reply:
I improved to go more towards your way, the results are now queriable with rpc `/block_results?height=0` as shown in the new PR description

I still can't figure that part out

> genesis txs are indexable by hash from any client (or the node's RPC)

- in the genesis, txs are passed in a gno.land type, so should I move genesis transaction from GnoGenesisState to InitChainRequest? If you have another idea to get the genesis transactions hash pls tell me
- I don't see how we could get the transaction hashes to fetch in the first place, this would require us to also expose a genesis block and I feel this is way beyond the scope of this PR